lib/IRGen/GenReflection.cpp

```diff
@@ -901,7 +908,7 @@ class CaptureDescriptorBuilder : public ReflectionMetadataBuilder {
 
     // Now add typerefs of all of the captures.
     for (auto CaptureType : CaptureTypes) {
-      addTypeRef(CaptureType);
+      addTypeRef(CaptureType, /*genericSignature*/nullptr);
```

Comment:
Should captures and captured parameters be using a particular signature?

Reply:
It looks like the capture type was computed with mapTypeOutOfContext(). The result of that is always canonical wrt. the generic signature that the archetype was built from.